numpy/core/src/umath/svml/svml_z0_acos_d_la.s

```diff
@@ -0,0 +1,2642 @@
+        movsd     -40(%rsp), %xmm11
+        movsd     -48(%rsp), %xmm13
+        subsd     %xmm13, %xmm11
+        addsd     4344+_vmldACosHATab(%rip), %xmm3
+        movsd     %xmm11, -40(%rsp)
+        movsd     -40(%rsp), %xmm14
+        mulsd     %xmm8, %xmm3
+        subsd     %xmm14, %xmm15
+        movsd     %xmm15, -48(%rsp)
+        movsd     -40(%rsp), %xmm10
+        movsd     -48(%rsp), %xmm9
+        addsd     %xmm9, %xmm4
+        addsd     4328+_vmldACosHATab(%rip), %xmm3
+        addsd     %xmm2, %xmm4
+        addsd     %xmm5, %xmm3
+        addsd     %xmm1, %xmm4
+        addsd     %xmm0, %xmm4
+        addsd     %xmm3, %xmm4
+        mulsd     %xmm6, %xmm4
+        mulsd     %xmm10, %xmm6
+        addsd     %xmm7, %xmm4
+        movsd     (%rdi), %xmm7
+        comisd    4184+_vmldACosHATab(%rip), %xmm7
+        ja        .LBL_2_13
+
+
+        movsd     4152+_vmldACosHATab(%rip), %xmm2
+        movaps    %xmm2, %xmm0
+        movsd     4160+_vmldACosHATab(%rip), %xmm5
+        subsd     %xmm6, %xmm0
+        subsd     %xmm4, %xmm5
+        movsd     %xmm0, -40(%rsp)
+        movsd     -40(%rsp), %xmm1
+        movsd     %xmm6, -56(%rsp)
+        subsd     %xmm1, %xmm2
+        subsd     %xmm6, %xmm2
+        movsd     %xmm2, -48(%rsp)
+        movsd     -40(%rsp), %xmm6
+        movsd     -48(%rsp), %xmm3
+        movaps    %xmm3, %xmm4
+        addsd     %xmm5, %xmm4
+
+.LBL_2_13:
+
+        addsd     %xmm4, %xmm6
+        movsd     %xmm6, (%rsi)
+        ret
+
+.LBL_2_14:
+
+        ucomisd   %xmm0, %xmm1
+        jp        .LBL_2_15
+        je        .LBL_2_19
+
+.LBL_2_15:
+
+        xorps     .L_2il0floatpacket.197(%rip), %xmm0
+        ucomisd   %xmm0, %xmm1
+        jp        .LBL_2_16
+        je        .LBL_2_18
+
+.LBL_2_16:
+
+        movl      $1, %eax
+        pxor      %xmm1, %xmm1
+        pxor      %xmm0, %xmm0
+        divsd     %xmm0, %xmm1
+        movsd     %xmm1, (%rsi)
+
+.LBL_2_17:
+
+        ret
+
+.LBL_2_18:
+
+        movsd     4152+_vmldACosHATab(%rip), %xmm0
+        addsd     4160+_vmldACosHATab(%rip), %xmm0
+        movsd     %xmm0, (%rsi)
+        ret
+
+.LBL_2_19:
+
+        movq      4184+_vmldACosHATab(%rip), %rdx
+        movq      %rdx, (%rsi)
+        ret
+
+.LBL_2_20:
+
+        divsd     %xmm1, %xmm1
+        movsd     %xmm1, (%rsi)
+        testl     $1048575, 4(%rdi)
+        jne       .LBL_2_17
+
+
+        cmpl      $0, (%rdi)
+        sete      %al
+        ret
+	.align    16,0x90
+
+	.cfi_endproc
+
+	.type	__svml_dacos_cout_rare_internal,@function
+	.size	__svml_dacos_cout_rare_internal,.-__svml_dacos_cout_rare_internal
+..LN__svml_dacos_cout_rare_internal.1:
+
+	.section .rodata, "a"
+	.align 64
+	.align 64
+__svml_dacos_data_internal:
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
+	.long	2147483648
+	.long	0
```

Comment:
Is it possible to use `.macro` directive similar to what GCC does? using macros to generate the constants or even the lookup tables can shrink the ASM files and also make the code easy to read. Another question, Why `.long` directive is used instead of `.quad`? 

Example for .macro directive to broadcast a vector:

```suggestion
double_vector __SIGNED_ZERO 0x8000000000000000
```
which `double_vector` is a macro defined as follows:
```C
.macro double_vector offset value
.if .-__svml_dacos_data_internal != \offset
.err
.endif
.rept 8
.quad \value
.endr
.endm
````

Reply:
That's true, I think using macro's can make it more readable. Considering that there are 44 files, it can take me a while to edit and ensure I don't break any algorithm. I think it might be better to do this one by one at a later time. SVML sources are planned to be part of a git submodule anyways and I can help maintain the module and make it more readable.

Reply:
Count on macros should be a later step after getting rid of bare offsets, and since it wouldn't be possible to get rid of them according to your/my comment above then we're facing the same situation "leave them as they are".